llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -709,6 +709,20 @@ Value *VPInstruction::generate(VPTransformState &State) {
   }
 }
 
+InstructionCost VPInstruction::computeCost(ElementCount VF,
+                                           VPCostContext &Ctx) const {
+  switch (getOpcode()) {
+  case VPInstruction::AnyOf: {
+    auto *VecI1Ty = VectorType::get(Type::getInt1Ty(Ctx.LLVMCtx), VF);
```

Comment:
Can we just infer the result type for the recipe and use that? And the `toVectorTy` helper for consistency with other ::comptueCost implementations?